ui/app/AppLayouts/Chat/components/ChannelContextMenu.qml

```diff
@@ -0,0 +1,143 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import "./"
+import "../../../../shared"
+import "../../../../imports"
+
+PopupMenu {
+    property int channelIndex
+    property bool channelMuted
+    property int chatType
+    property string chatName
+    property string chatId
+    property string chatIdenticon
+    property var groupInfoPopup
+    property var groupsListView
+
+    id: channelContextMenu
+    width: 175
+    subMenuIcons: [
+        /* { */
+        /*     source:  Qt.resolvedUrl("../../../img/bell.svg"), */
+        /*     width: 16, */
+        /*     height: 16 */
+        /* }, */
+        {
+            source: Qt.resolvedUrl("../../../img/fetch.svg"),
+            width: 16,
+            height: 16
+        }
+    ]
+
+    function openMenu(channel, index) {
+        channelContextMenu.channelMuted = channel.muted
+        channelContextMenu.chatType = channel.chatType
+        channelContextMenu.chatName = channel.name
+        channelContextMenu.chatId = channel.id
+        channelContextMenu.chatIdenticon = channel.identicon
+        if (index !== undefined) {
+            channelContextMenu.channelIndex = index
+        }
+        channelContextMenu.popup()
+    }
+
+    Action {
+        enabled: channelContextMenu.chatType !== Constants.chatTypePublic
+        text: {
+            if (channelContextMenu.chatType === Constants.chatTypeOneToOne) {
+                //% "View Profile"
+                return qsTrId("view-profile")
+            }
+            if (channelContextMenu.chatType === Constants.chatTypePrivateGroupChat) {
+                //% "View Group"
+                return qsTrId("view-group")
+            }
+            //% "Share Chat"
+            return qsTrId("share-chat")
+        }
+        icon.source: "../../../img/group.svg"
+        icon.width: 16
+        icon.height: 16
+        onTriggered: {
+            //chatsModel.setActiveChannelByIndex(channelContextMenu.channelIndex)
```

Comment:
This is a left over but needs some thought:

Right now, what thiis does, is it possibly changes the current active channel.
This means, when you click e.g. "View Profile" on a channel item that is not the active channel at the moment, it will jump to that channel.

We probably don't want that.